crypto/src/main/java/net/corda/v5/crypto/DigestAlgorithmName.java

```diff
@@ -0,0 +1,83 @@
+package net.corda.v5.crypto;
+
+import java.util.Locale;
+import net.corda.v5.base.annotations.CordaSerializable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The digest algorithm name. This class is to be used in Corda hashing API.
+ *
+ * @param name The name of the digest algorithm to be used for the instance.
+ */
+
+@CordaSerializable
+public final class DigestAlgorithmName {
+    @NotNull private final String name;
+    /**
+     * Instance of SHA-256
+     */
+    @NotNull public static final DigestAlgorithmName SHA2_256 = new DigestAlgorithmName("SHA-256");
+
+    /**
+     * Instance of Double SHA-256
+     */
+    @NotNull public static final DigestAlgorithmName SHA2_256D = new DigestAlgorithmName("SHA-256D");
+
+    /**
+     * Instance of SHA-384
+     */
+    @NotNull public static final DigestAlgorithmName SHA2_384 = new DigestAlgorithmName("SHA-384");
+
+    /**
+     * Instance of SHA-512
+     */
+    @NotNull public static final DigestAlgorithmName SHA2_512 = new DigestAlgorithmName("SHA-512");
+
+
+    /**
+     * Instance of algorithm which is considered to be default. Set as SHA-256
+     */
+    @NotNull public static final DigestAlgorithmName DEFAULT_ALGORITHM_NAME;
+
+
+    /**
+     * Converts a [DigestAlgorithmName] object to a string representation.
+     */
+    @NotNull public String toString() {
+        return this.name;
+    }
+    
+    /**
+     * Returns a hash code value for the object.
+     */
+    public int hashCode() {
+        return this.name.toUpperCase(Locale.ROOT).hashCode();
```

Comment:
No sure what `Locale.ROOT` is. Is it needed?

Reply:
It seems we do not need it, so removed.